marketplacetests/payment/app.py

```diff
@@ -88,14 +95,18 @@ def wait_for_buy_app_section_displayed(self):
         self.wait_for_element_displayed(*self._buy_button_locator)
 
     def tap_buy_button(self):
+        self._tap_payment_button(self._buy_button_locator)
+
+    def tap_cancel_button(self):
+        self._tap_payment_button(self._cancel_button_locator)
+
+    def _tap_payment_button(self, button_locator):
         self.marionette.switch_to_frame()
         self.wait_for_element_not_displayed(*self._loading_throbber_locator)
         payment_iframe = self.marionette.find_element(*self._payment_frame_locator)
         self.marionette.switch_to_frame(payment_iframe)
-        if self.is_element_present(*self._buy_button_locator):
-            self.marionette.find_element(*self._buy_button_locator).tap()
-        else:
-            self.marionette.find_element(*self._in_app_confirm_buy_button_locator).tap()
+        self.wait_for_element_displayed(*button_locator)
+        self.marionette.find_element(*button_locator).tap()
         self.marionette.switch_to_frame()
         self.wait_for_element_not_present(*self._payment_frame_locator)
-        self.apps.switch_to_displayed_app()
+        self.switch_to_marketplace_frame()
```

Comment:
Yes, exactly. I thought about doing that, but it seemed messy to have two different copies of the Marketplace app in the request. Also, the `__init__` method for `Marketplace` expects an `app_name` argument that I didn't want to have to worry about passing around. Subclassing the `Marketplace` app will solve this problem anyway.